tests/layer/test_appnp.py

```diff
@@ -0,0 +1,76 @@
+from stellargraph.layer.appnp import *
```

Comment:
You have no tests for edge cases. What if the generator is of the wrong type? What if approx_iter is a negative number or a float? What if layer_sizes is of different length than activations. What if the last layer's size is not equal to the number of classes to predict,  and so on?

You need to add more tests!